sympy/sets/sets.py

```diff
@@ -1438,6 +1438,8 @@ def reduce(args):
                              if all(other.contains(x) == True for other in other_args)])
                 unk = [x for x in s
                        if any(other.contains(x) not in (True, False) for other in other_args)]
+                other_args = [elm - FiniteSet(*unk) if elm.is_FiniteSet and
+                                not FiniteSet(*unk).is_EmptySet else elm for elm in other_args]
```

Comment:
> Under which conditions is FiniteSet(*unk).is_EmptySet true?

if for every element in `FiniteSet`(here `s`), we can determine whether it belongs to every other `FiniteSet ` or `Interval` then `FiniteSet(*unk)` is EmptySet.

> Is it not equivalent to unk being empty?

Yes both are equivalent.